### PR TITLE
[To Discuss] Add header's `label/titlePadding`

### PR DESCRIPTION
**Pull request by kanitw**
Depends on https://github.com/vega/vega-lite/pull/3851

### CODE REVIEW

File: src/compile/layout/header.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// TODO: refactor this into proper parse/assemble phase like other
```
Comment (domoritz) [inline src/compile/layout/header.ts]: like other what?

File: src/facet.ts [hunk 2]
```diff
@@ -17,7 +17,21 @@ export interface Header extends Guide {
    */
   labelAngle?: number;
 
-  // TODO: labelPadding
+  /**
+   * Padding spacing in pixels between header labels and axes.
```
Comment (domoritz) [inline src/facet.ts]: padding? spacing? Why both?

Comment (domoritz) [reply src/facet.ts]: See https://vega.github.io/vega-lite/docs/axis.html#title for how we name things in axes.

Comment (kanitw) [reply src/facet.ts]: I already saw that.  This is pretty consistent.

File: src/compile/layout/header.ts (hunk 1 above)
Comment (kanitw) [reply src/compile/layout/header.ts]: Like axis/legend.

**Comment by kanitw** [comment]
Thinking more about this, let's not add this until someone asks for this, or until someone think clearly how `header` would work with wrapped facet so we can avoid breaking changes.  

This isn't as important as spacing anyway.

**Comment by kanitw** [comment]
#4028 is probably better (simpler code)

**Pull request closed without merging**